llvm/test/MC/LoongArch/Basic/Privilege/invalid.s

```diff
@@ -3,9 +3,9 @@
 
 ## csrxchg: rj != 0,1
 csrxchg $a0, $zero, 0
-# ERR: :[[#@LINE-1]]:15: error: must not be $r0 or $r1
+# ERR: :[[#@LINE-1]]:15: error: invalid operand for instruction
```

Comment:
You can have this:

```diff
diff --git a/llvm/lib/Target/LoongArch/AsmParser/LoongArchAsmParser.cpp b/llvm/lib/Target/LoongArch/AsmParser/LoongArchAsmParser.cpp
index 39c5e034f2a4..81e8c3657d5c 100644
--- a/llvm/lib/Target/LoongArch/AsmParser/LoongArchAsmParser.cpp
+++ b/llvm/lib/Target/LoongArch/AsmParser/LoongArchAsmParser.cpp
@@ -1633,6 +1633,9 @@ LoongArchAsmParser::validateTargetOperandClass(MCParsedAsmOperand &AsmOp,
     return Match_Success;
   }
 
+  if (Kind == MCK_GPRNoR0R1 && (Reg == LoongArch::R0 || Reg == LoongArch::R1))
+    return Match_RequiresOpnd2NotR0R1;
+
   return Match_InvalidOperand;
 }
 
```

So that we can have the nice diagnostics here intact.

Reply:
Good idea.